### PR TITLE
EXAMPLE: Refactor: UpdateUi.tscn UpdateUi.gd

### PR DESCRIPTION
**Pull request by TheDuriel**
Provides and initial basis for the UpdateUI refactor.

Several TODOs are left.

See https://github.com/moonwards1/Moonwards-Virtual-Moon/issues/59 for more information

### CODE REVIEW

File: core/update/scenes/UpdateUI.tscn [hunk 1]
```diff
@@ -5,12 +5,12 @@
 [sub_resource type="StyleBoxFlat" id=1]
 bg_color = Color( 0, 0, 0, 0.760784 )
 
-[node name="MainMenu" type="Control"]
+[node name="UpdateUI" type="Control"]
```
Comment (MolbOrg) [inline core/update/scenes/UpdateUI.tscn]: MainMenu name is a hack for to be able to run it for tests as a standalone scene, as it needs to attach to the main tree to work with the network notifications and such, and the main scene at the run is MainMenu  root

File: core/update/scripts/UpdateUI.gd [hunk 2]
```diff
@@ -1,154 +1,175 @@
...
+	set_control_text(_serverstatus_label, "online")
+
+
+func _on_server_offline() -> void:
+	set_control_text(_serverstatus_label, "offline")
+	_disconnect_signals()
+
+
+	# TODO: Confirm type. Bool? Or status code?
+func _on_client_protocol(state) -> void:
```
Comment (MolbOrg) [inline core/update/scripts/UpdateUI.gd]: does not belong here, it is not used nor needed in UpdateUI

File: core/update/scripts/UpdateUI.gd [hunk 3]
```diff
@@ -1,154 +1,175 @@
...
+	_connect_signals()
+	
 	options.Updater.ClientOpenConnection()
-	var l = $VBoxContainer/ClientStatus/StartUpdate
-	set_label(l, "processing")
+	
+	set_control_text(_update_button, "processing")
+	
 	var res = options.Updater.ui_ClientUpdateData()
+	# What?
```
Comment (MolbOrg) [inline core/update/scripts/UpdateUI.gd]: returns an error(false) if client needs an update, or some another error happened. If thie frontend gets as user main element, needs some notification or else to process that situation. As debug element does not need that, but a place holder to indicate there is something, if needed

**Comment by TheDuriel** [comment]
Closing this as the example is quite out of date.

**Pull request closed without merging**